src/Shared/PackageManagers/NuGetProjectManager.cs

```diff
@@ -208,77 +145,26 @@ public override async Task<bool> PackageExists(PackageURL purl, bool useCache =
                 Logger.Trace("Provided PackageURL was null.");
                 return false;
             }
-            string packageName = purl.Name;
-            HttpClient httpClient = CreateHttpClient();
 
-            return await CheckJsonCacheForPackage(httpClient, $"{RegistrationEndpoint}{packageName.ToLowerInvariant()}/index.json", useCache);
+            return await _nuGetPackageActions.DoesPackageExistAsync(purl, useCache: useCache);
         }
 
         /// <inheritdoc />
         public override async Task<IEnumerable<string>> EnumerateVersions(PackageURL purl, bool useCache = true)
         {
-            Logger.Trace("EnumerateVersions {0}", purl?.ToString());
+            Logger.Trace("EnumerateVersions {0}", purl.ToString());
 
-            if (purl == null || purl.Name is null)
+            if (string.IsNullOrWhiteSpace(purl.Name))
             {
                 return new List<string>();
```

Comment:
```suggestion
                return Array.Empty<string>();
```
Less allocation overhead than providing a list.